### PR TITLE
DDP-5627 Script to bulk withdraw a list of participants

### PR DESCRIPTION
**Pull request by zyme**
## Context

To withdraw a lot of participants, I made a script for DDP-5267.  The script issues a withdrawal in DSM, updates the enrollment status in DSS, and requests an ES update along the way.

## Checklist

- [ ] I have labeled the type of changes involved using the `C-*` labels.
- [ ] I have assessed potential risks and labeled using the `R-*` labels.
- [ ] I have considered error handling and alerts, and added `L-*` labels as needed.
- [ ] I have considered security and privacy, and added `I-*` labels as needed
- [ ] I have analyzed my changes for stability, fault tolerance, graceful degradation, performance bottlenecks and written a brief summary in this PR.
- [ ] If applicable, I have discussed the analytics needs at both a platform and study level with Product and instrumented code accordingly.
- [ ] If applicable, my UI/UX changes have passed muster with Product/Design via an over-the-shoulder review, screenshots, etc.

_If unsure or need help with any of the above items, add the `help wanted` label. For items that starts with `If applicable`, if it is not applicable, check it off and add `n/a` in front._

## FUD Score

_Overall, how are you feeling about these changes?_

- [ ] :relaxed: All good, business as usual!
- [ ] :sweat_smile: There might be some issues here
- [ ] :scream: I'm sweaty and nervous

## How do we demo these changes?

_How does one observe these changes in a deployed system? Note that **user visible** encompasses many personas--not just patients and study staff, but also ops duty, your fellow devs, compliance, etc._

- [ ] They are user-visible in dev as a regular user journey and require no additional instructions.
- [ ] Getting dev into a state where this is user-visible requires some tech fiddling. I have documented these steps in the related ticket.
- [ ] Requires other features before it's human visible. I have documented the blocking issues in jira.
- [ ] I have no idea how to demo this. Please help me!

## Testing

- [ ] I have written automated positive tests
- [ ] I have written automated negative tests
- [ ] I have written zero automated tests but have poked around locally to verify proper functionality
- [ ] The jira ticket has acceptance criteria and QA has the needed information to test changes

## Release

- [ ] These changes require no special release procedures--just code!
- [ ] Releasing these changes requires special handling and I have documented the special procedures in the release plan document

### CODE REVIEW

File: pepper-apis/src/main/java/org/broadinstitute/ddp/util/BulkWithdrawCLI.java [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+                        "values(?,?,?,?,?);");
+
+
+                for (String hruid : hruids) {
+                    LOG.info("Withdrawing {}", hruid);
+                    User participant = userDao.findUserByHruid(hruid).get();
+                    enrollmentDao.insert(participant.getGuid(), studyGuid, EnrollmentStatusType.EXITED_AFTER_ENROLLMENT, now.toEpochMilli());
+                    dsmWithdrawStmt.setLong(1, instanceId);
+                    dsmWithdrawStmt.setString(2, participant.getGuid());
+                    dsmWithdrawStmt.setLong(3, now.toEpochMilli());
```
Comment (SimoneMaiwald) [inline pepper-apis/src/main/java/org/broadinstitute/ddp/util/BulkWithdrawCLI.java:82]: should be millisec not epoch

File: pepper-apis/src/main/java/org/broadinstitute/ddp/util/BulkWithdrawCLI.java [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+                instanceStmt.setString(1, studyGuid);
+                ResultSet resultSet = instanceStmt.executeQuery();
+                resultSet.next();
+                long instanceId = resultSet.getLong(1);
+                if (resultSet.next()) {
+                    throw new RuntimeException("Too many row rows for instance " + instanceId);
+                }
+
+
+                PreparedStatement dsmWithdrawStmt = handle.getConnection().prepareStatement("insert into dev_dsm_db.ddp_participant_exit(ddp_instance_id, ddp_participant_id,exit_date, exit_by,in_ddp)\n" +
```
Comment (SimoneMaiwald) [inline pepper-apis/src/main/java/org/broadinstitute/ddp/util/BulkWithdrawCLI.java:72]: "dev_dsm_db" is still in there

File: pepper-apis/src/main/java/org/broadinstitute/ddp/util/BulkWithdrawCLI.java [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+                new TransactionWrapper.DbConfiguration(TransactionWrapper.DB.APIS, maxConnections, dbUrl));
+        Instant now = Instant.now();
+
+        try {
+            TransactionWrapper.useTxn(handle -> {
+                JdbiUserStudyEnrollment enrollmentDao = handle.attach(JdbiUserStudyEnrollment.class);
+                UserDao userDao = handle.attach(UserDao.class);
+
+                DataExportDao exportDao = handle.attach(DataExportDao.class);
+                PreparedStatement instanceStmt = handle.getConnection().prepareStatement("select ddp_instance_id from dev_dsm_db.ddp_instance where instance_name= ?");
```
Comment (SimoneMaiwald) [inline pepper-apis/src/main/java/org/broadinstitute/ddp/util/BulkWithdrawCLI.java:62]: "dev_dsm_db" is still in there

Comment (zyme) [reply pepper-apis/src/main/java/org/broadinstitute/ddp/util/BulkWithdrawCLI.java:62]: Yep, better safe than sorry for a PR :)

**Comment by yufengwng** [comment]
Is there an API in DSM that we can call to withdraw a participant? Once we call that, DSM should in turn call Pepper's withdraw participant API like usual, which will handle withdrawing and updating ES cleanly.

**Comment by Trimax** [comment]
Discussed with @zyme. Not going to be merged.

**Pull request closed without merging**